kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/NonJVMClientsST.java

```diff
@@ -0,0 +1,113 @@
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithNoFilters(clusterName, 1);
+        String bootstrap = kroxylicious.getBootstrap();
+
+        LOGGER.info("And a kafka Topic named {}", topicName);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+
+        LOGGER.info("When {} messages '{}' are sent to the topic '{}'", numberOfMessages, MESSAGE, topicName);
+        KroxyliciousSteps.produceMessagesWithKcat(namespace, topicName, bootstrap, MESSAGE, numberOfMessages);
+
+        LOGGER.info("Then the {} messages are consumed", numberOfMessages);
+        String result = KroxyliciousSteps.consumeMessagesWithKcat(namespace, topicName, bootstrap, expectedMessage, Duration.ofMinutes(2));
+        LOGGER.info("Received: " + result);
+
+        assertThat(result).withFailMessage("expected message have not been received!").contains(expectedMessage);
+    }
+
+    @Test
+    void kafkaGoClients(String namespace) {
+        int numberOfMessages = 1;
+        String expectedMessage = MESSAGE + "-" + (numberOfMessages);
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithNoFilters(clusterName, 1);
+        String bootstrap = kroxylicious.getBootstrap();
+
+        LOGGER.info("And a kafka Topic named {}", topicName);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+
+        LOGGER.info("When the messages '{}' are sent to the topic '{}'", MESSAGE, topicName);
+        KroxyliciousSteps.produceMessagesWithKafkaGo(namespace, topicName, bootstrap);
+
+        LOGGER.info("Then the messages are consumed");
+        String result = KroxyliciousSteps.consumeMessagesWithKafkaGo(namespace, topicName, bootstrap, expectedMessage, Duration.ofMinutes(2));
+        LOGGER.info("Received: " + result);
+
+        assertThat(result).withFailMessage("expected message have not been received!").contains(expectedMessage);
+    }
+
+    /**
+     * Sets before all.
+     */
+    @BeforeAll
+    void setupBefore() {
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(Constants.KAFKA_DEFAULT_NAMESPACE, clusterName);
+        if (!kafkaPods.isEmpty()) {
+            LOGGER.warn("Skipping kafka deployment. It is already deployed!");
+            return;
+        }
+        LOGGER.info("Deploying Kafka in {} namespace", Constants.KAFKA_DEFAULT_NAMESPACE);
+
+        Kafka kafka = KafkaTemplates.kafkaPersistentWithKRaftAnnotations(Constants.KAFKA_DEFAULT_NAMESPACE, clusterName, 3).build();
+
+        resourceManager.createResourceWithWait(
+                KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(BROKER_NODE_NAME, kafka, 3).build(),
+                kafka);
+    }
```

Comment:
Probably something for a separate PR but I think this should the be the responsibility of a junit extension that ensures there is a Kafka cluster available for tests to use rather than reproducing this logic in every test.

Reply:
The problem doing this is that we can have different configs for kafka depending on the Test Suite. That's why we have included the kafka creation inside each test suite.